### DISCUSSION TITLE
Is there a way to specify which columns should be available on the list view?

### DISCUSSION THREAD

**Discussion by ninahorne** [payloadcms/payload#516] [Q&A]
On the list view for a collection, I would like to limit the amount of columns a user can select to display. I would also like to specify which fields are filterable. I mostly want to do this to streamline/simplify the admin panel for certain collections. Or possibly even hide the columns / filter buttons for some collections. Is this possible? I didn't see anything in the docs about it.

**Comment by DanRibbens** [comment]
We don't currently have a way to limit the filter or list columns available, though you can set the default columns to show in the List view for a collection, see `admin.defaultColumns`. https://payloadcms.com/docs/configuration/collections#admin-options

I think we should add properties that control list and filters to the field `admin` config for the admin UI List component to use.

As a workaround you could make a custom List component for the collections you want to have more control over. From there you copy the existing List component from the source and add your own filter for field names to make them not appear in the columns / filters. As I write this out I'm realizing this may be more work than just adding the feature to Payload...
You would copy the Default list also and customize that: https://github.com/payloadcms/payload/blob/master/src/admin/components/views/collections/List/Default.tsx
You could pass different options in to the ListControls component or replace it if you want to not even show filters.

There is a lot you could do with your own components, but what you're describing should be build in to the Payload configuraition long term.

I'd be happy to prioritize these features or review and test any PRs that come in.
Thanks!

**Reply by wherewego-nina** [reply]
@DanRibbens  Would you possibly be able to explain to me why this code doesn't work to remove certain fields from the Tables? I tried to look through the source code it wasn't clear to me why this wouldn't work

```
import React, { useEffect, useState } from 'react';
import List from 'payload/dist/admin/components/views/collections/List/Default'
import { Props } from 'payload/dist/admin/components/views/collections/List/types'


const RegionsListView: React.FC<Props> = (props) => {
  const [updatedProps, setUpdatedProps] = useState<Props | null>(null)

  useEffect(() => {
    const fields = props.collection.fields.filter((f: any) => f.type != 'ui' && f?.name != 'geoJSONRaw');
    props.collection.fields = fields;
    if (props.data.docs) {

      props.data.docs = props.data.docs.map(doc => {
        const copyDoc = { ...doc };
        delete copyDoc.geoJSONRaw;
        return copyDoc;
      })
    }
    setUpdatedProps({ ...props })
  }, [props])


  if (!updatedProps) return <></>

  return (
    <List {...updatedProps} />
  );
};

export default RegionsListView;


```

The props are properly being updated but the table is unaffected. Does it only

**Reply by DanRibbens** [reply]
I can't say by looking at it and not experimenting with it myself. I'm not able to do that right now, so I think adding the feature to Payload as something you can manage in the field config instead would be more worthwhile if somebody is able to take this on.

**Comment by ninahorne** [comment]
Thank you for the response. Yeah, for the time being I'll create a custom component using payload source code. It's definitely a lot to manage, but it will work.  It def would be awesome to have this as part of the config. it would be nice to be able to search more than one field as well.

**Comment by Stupidism** [comment]
@DanRibbens  I'd like create a PR to add an option like `field.admin.disabled: { list: true, update: false, create: false }`?

Anything else to pay attention to except this [contributing guideline](https://github.com/payloadcms/payload/blob/master/contributing.md)?

**Reply by DanRibbens** [reply]
We have to think about this a little. Fields can already be defined with `admin.disabled` with a boolean that when true ~makes the input disabled in the edit view~ completely takes it out of the admin UI.
We're talking about a few different things here.
1. Disabling fields from columns in a list
2. Disabling fields from being filterable in admin UI
3. Disabling fields within create and update

I like your idea of making the disabled an optional object. That is currently a boolean that removes the field everywhere in the admin. We'd want to make sure that the changes allow for that to still work. The properties would need to be:
```
disabled: { 
  filter?: boolean
  column?: boolean
  create?: boolean
  update?: boolean
  }
```

All of these would default to false. For globals, since they don't have a separate create/update it would be determined just by the `update`.

Does this make sense? Let me know if you have questions or need any guidance.

Appreciate your willing to jump into this @Stupidism!

Edit: Corrected point about `admin.disabled` and formatting.

**Comment by m-grgic** [comment]
@DanRibbens Was wondering if there were any advancements on this issue.
The proposed solution I think is ideal. We're then given the ability to hide a field from the API and (given the above) hide the field on views where the it can be displayed within the admin panel. Every control case about the visibility of the field seems covered.
We're currently exploring options on how to achieve column hiding, needless to say this discussion had been a great resource, but of course having a built in option that enables the feature seems great. Let me know, would love to contribute

**Comment by tyteen4a03** [comment]
Any updates on this?

**Reply by BrianJM** [reply]
I'm also checking the status and watching [this](https://github.com/payloadcms/payload/discussions/516#discussioncomment-3519662).

**Comment by DanRibbens** [comment]
If anyone has a new year's resolution is to contribute more to open-source, this would be a great enhancement to work on ;)

I'd gladly review it when it is opened.

**Comment by franciscolourenco** [comment]
This has been added in [v2.16.0](https://github.com/payloadcms/payload/releases/tag/v2.16.0) ✅ 🎈 

- `admin.hidden` hides a field in the edit view
- `admin.disableListColumn` hides a field in the list view columns
- `admin.disableListFilter ` hides a field in the filters 

[Docs](https://payloadcms.com/docs/fields/overview#admin-config)
